app/docker/networks/edit/NetworkContainersTable.tsx

```diff
@@ -0,0 +1,91 @@
+import { Widget, WidgetBody, WidgetTitle } from '@/portainer/components/widget';
+import { DetailsTable } from '@/portainer/components/DetailsTable';
+import { Button } from '@/portainer/components/Button';
+import { Authorized } from '@/portainer/hooks/useUser';
+import { EnvironmentId } from '@/portainer/environments/types';
+import { Link } from '@/portainer/components/Link';
+
+import { NetworkContainer, NetworkId } from '../types';
+import { useDisconnectContainer } from '../queries';
+
+type Props = {
+  networkContainers: NetworkContainer[];
+  nodeName: string;
+  environmentId: EnvironmentId;
+  networkId: NetworkId;
+};
+
+const tableHeaders = [
+  'Container Name',
+  'IPv4 Address',
+  'IPv6 Address',
+  'MacAddress',
+  'Actions',
+];
+
+export function NetworkContainersTable({
+  networkContainers,
+  nodeName,
+  environmentId,
+  networkId,
+}: Props) {
+  const disconnectContainer = useDisconnectContainer(environmentId, networkId);
+
+  return (
+    <div className="row">
+      <div className="col-lg-12 col-md-12 col-xs-12">
+        <Widget>
+          <WidgetTitle title="Containers in network" icon="fa-server" />
+          <WidgetBody className="nopadding">
+            <DetailsTable
+              headers={tableHeaders}
+              dataCy="networkDetails-networkContainers"
+            >
+              {networkContainers.map((container) => (
+                <tr key={container.Id}>
+                  <td>
+                    <Link
+                      to="docker.containers.container"
+                      params={{
+                        id: container.Id,
+                        nodeName,
+                      }}
+                      title={container.Name}
+                    >
+                      {container.Name}
+                    </Link>
+                  </td>
+                  <td>{container.IPv4Address || '-'}</td>
+                  <td>{container.IPv6Address || '-'}</td>
+                  <td>{container.MacAddress || '-'}</td>
+                  <td>
+                    <Authorized authorizations="DockerNetworkDisconnect">
+                      <Button
+                        dataCy={`networkDetails-disconnect${container.Name}`}
+                        size="xsmall"
+                        color="danger"
+                        onClick={() => {
+                          if (container.Id) {
```

Comment:
why would Id be undefined?

Reply:
The id is the key for each container in the containers section of the network response. 

In the containers response its within each container object.

I could either 
*create two types, 
*transform the containers object within the network response
*keep the existing changes

Reply:
if `id` can be undefined then L49 is error prone... id has to be defined

Reply:
this is still relevant.

Reply:
I updated the types to differentiate the containers in the network response from the containers in the container response, to avoid any possible undefined issues